pkg/tfshim/sdk-v2/provider2.go

```diff
@@ -168,7 +239,9 @@ func (p *planResourceChangeImpl) Diff(
 	if err != nil {
 		return nil, err
 	}
-	cfg, err := recoverAndCoerceCtyValueWithSchema(res.CoreConfigSchema(), config.Config)
+
+	cfg, err := recoverAndCoerceCtyValueWithSchema(res.CoreConfigSchema(),
+		p.configWithTimeouts(ctx, config.Config, opts.TimeoutOptions, ty))
```

Comment:
It looks like this is only called during `Diff`. Doesn't this need to be called during update as well?